pkg/metrics/cluster/config.go

```diff
@@ -50,8 +51,10 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
 // FlagSet with a specified prefix.
 func (c *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.BoolVar(&c.Enabled, prefix+"enabled", false, "enables the scraping service mode")
-	f.DurationVar(&c.ReshardInterval, prefix+"reshard-interval", time.Minute*1, "how often to manually reshard")
-	f.DurationVar(&c.ReshardTimeout, prefix+"reshard-timeout", time.Second*30, "timeout for cluster-wide reshards and local reshards. Timeout of 0s disables timeout.")
+	f.DurationVar(&c.ReshardInterval, prefix+"reshard-interval", time.Minute*1, "how often to manually refresh configuration")
+	f.DurationVar(&c.ReshardTimeout, prefix+"reshard-timeout", time.Second*30, "timeout for refreshing the configuration. Timeout of 0s disables timeout.")
+	f.DurationVar(&c.RefreshTimeout, prefix+"refresh-timeout", time.Second*30, "timeout for the cluster reshard. Timeout of 0s disables timeout.")
```

Comment:
I think we can set this a lot smaller since it kicks off an async job. 5s maybe?

Reply:
I am a bit hesitant to drop that low. 30 seconds is likely overkill but is inline with our previous values.